client/src/app/routes/stardust/OutputsList.tsx

```diff
@@ -0,0 +1,117 @@
+/* eslint-disable @typescript-eslint/no-floating-promises */
+import { IOutputResponse } from "@iota/iota.js-stardust";
+import React, { useEffect, useState } from "react";
+import { RouteComponentProps, useLocation } from "react-router-dom";
+import { ServiceFactory } from "../../../factories/serviceFactory";
+import { STARDUST } from "../../../models/config/protocolVersion";
+import { StardustTangleCacheService } from "../../../services/stardust/stardustTangleCacheService";
+import Pagination from "../../components/Pagination";
+import Spinner from "../../components/Spinner";
+import Output from "../../components/stardust/Output";
+import "./OutputsList.scss";
+import OutputsListProps from "./OutputsListProps";
+
+interface OutputsListState {
+    outputIds: string[];
+    tag: string;
+}
+
+interface OutputsListResponse {
+    output: IOutputResponse;
+    outputId: string;
+}
+
+const TOKEN_PAGE_SIZE: number = 5;
+
+const OutputsList: React.FC<RouteComponentProps<OutputsListProps>> = (
+    { match: { params: { network } } }
+) => {
+    const [outputsDetail, setOutputsDetail] = useState<OutputsListResponse[] | undefined>();
+    const { state } = useLocation<OutputsListState>();
+    const [currentPage, setCurrentPage] = useState<OutputsListResponse[]>([]);
+    const [pageNumber, setPageNumber] = useState(1);
+    const [isLoading, setIsLoading] = useState(true);
+    const { outputIds, tag } = state ?? {
+        outputIds: [],
+        tag: ""
+    };
+
+    useEffect(() => {
+        if (outputIds.length > 0) {
+            getOutputDetails(outputIds);
+        } else {
+            setIsLoading(false);
+        }
+    }, []);
+
+    useEffect(() => {
+        const from = (pageNumber - 1) * TOKEN_PAGE_SIZE;
+        const to = from + TOKEN_PAGE_SIZE;
+        if (outputsDetail) {
```

Comment:
If `outputDetails` initialises with empty array, you can avoid this if, just slice whatever is there.

Reply:
Done.